device/src/lib.rs

```diff
@@ -2,16 +2,16 @@
 #![cfg_attr(feature = "async", feature(async_fn_in_trait))]
 #![allow(incomplete_features)]
 
+extern crate alloc;
```

Comment:
This should be behind a feature flag for the cases where alloc is not supported (most embassy examples are no alloc), but I also can't see any place in the code that requires it?

Reply:
Woah! Thanks for catching that. It's really not meant to be there. My editor does that sometimes and now I'm motivated to disable that feature.